src/modules/hip/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,1439 @@
+
+    __syncthreads();                                                                // syncthreads after Shared load
+
+    // Reduction of 16 floats on 16 threads per block in x dimension (for every y dimension)
+    for (int threadMax = 8; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRSumSharedRowPtr[hipThreadIdx_x] += partialRSumSharedRowPtr[hipThreadIdx_x + threadMax];
+            partialGSumSharedRowPtr[hipThreadIdx_x] += partialGSumSharedRowPtr[hipThreadIdx_x + threadMax];
+            partialBSumSharedRowPtr[hipThreadIdx_x] += partialBSumSharedRowPtr[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    if (hipThreadIdx_x == 0)
+    {
+        // Reduction of 16 floats on 16 threads per block in y dimension
+        for (int threadMax = 8, increment = 128; threadMax >= 1; threadMax /= 2, increment /= 2)
+        {
+            if (hipThreadIdx_y < threadMax)
+            {
+                partialRSumSharedRowPtr[0] += partialRSumSharedRowPtr[increment];
+                partialGSumSharedRowPtr[0] += partialGSumSharedRowPtr[increment];
+                partialBSumSharedRowPtr[0] += partialBSumSharedRowPtr[increment];
+            }
+            __syncthreads();
+        }
+
+        // Final store to dst
+        if (hipThreadIdx_y == 0)
+        {
+            int idx = ((hipBlockIdx_z * hipGridDim_y + hipBlockIdx_y) * hipGridDim_x + hipBlockIdx_x) * 3;
+            tensorSumArr[idx] = partialRSumSharedRowPtr[0];
+            tensorSumArr[idx + 1] = partialGSumSharedRowPtr[0];
+            tensorSumArr[idx + 2] = partialBSumSharedRowPtr[0];
+        }
+    }
+}
+
+// -------------------- Set 2 - Kernel Executors --------------------
+//Handle U8 datatype
+RppStatus hip_exec_tensor_sum(Rpp8u *srcPtr,
+                              RpptDescPtr srcDescPtr,
+                              Rpp64u *tensorSumArr,
+                              RpptROIPtr roiTensorPtrSrc,
+                              RpptRoiType roiType,
+                              rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int localThreads_x = LOCAL_THREADS_X;
+    int localThreads_y = LOCAL_THREADS_Y;
+    int localThreads_z = LOCAL_THREADS_Z;
+    int globalThreads_x = (srcDescPtr->w + 7) >> 3;
+    int globalThreads_y = srcDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
+    int gridDim_x = (int) ceil((float)globalThreads_x/localThreads_x);
+    int gridDim_y = (int) ceil((float)globalThreads_y/localThreads_y);
+    int gridDim_z = (int) ceil((float)globalThreads_z/localThreads_z);
+
+    if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+    {
+        Rpp32u partialSumArrLength = gridDim_x * gridDim_y * gridDim_z;
+        Rpp32u *partialSumArr;
+        hipMalloc(&partialSumArr, sizeof(Rpp32u) * partialSumArrLength);
```

Comment:
Pls use the preallocated handle memory for this, and remove all hipMalloc and hipFree from here

Reply:
Introduced 
hipMemRpp32u uintReductionArr;
hipMemRpp32s intReductionArr;

in rppdefs.h and all other places and set this to size similar to maskArr(3840x2160)